nhs/utils/logging.py

```diff
@@ -36,19 +37,46 @@ def emit(self, record: logging.LogRecord) -> None:
         )
 
 
-def config_logger() -> None:
+def read_config(config_path: str = "configurations.yml") -> Dict[str, Any]:
     """
-    Configure loguru logger settings and set it as as the default logger
+    Read YAML configuration file and return as a dictionary.
+
+    Args:
+        config_path (str): Path to the configuration file. Defaults to 'configurations.yml'.
+
+    Returns:
+        Dict[str, Any]: Configuration settings as a dictionary.
+
+    Raises:
+        FileNotFoundError: If the configuration file is not found.
+        yaml.YAMLError: If there's an error parsing the YAML file.
+    """
+    try:
+        with open(config_path, "r") as file:
+            config: Dict[str, Any] = yaml.safe_load(file)
+        logger.info(f"Configuration loaded from {config_path}")
+        return config
+    except FileNotFoundError:
+        logger.error(f"Configuration file not found: {config_path}")
+        raise
+    except yaml.YAMLError as e:
+        logger.error(f"Error parsing configuration file: {e}")
+        raise
```

Comment:
ditto, move to `config.py`

Reply:
Also, use `logger.catch(reraise=True)` if you don't want the program to continue. use 1logger.catch()` if the program should continue as normal

Reply:
I think we should definitely terminate if logger isn't working, but I can't seem to get logger.catch to work.

Is logger.error not going to terminate the program anyway?